### PR TITLE
Use ProxyGet to proxy requests to the Catalog server

### LINKED ISSUE #15: Add support for client-go's ProxyGet mechanism
**What feature/behavior/change do you want?**

Instead of relying on the user to setup a port-forward to access the API, use a [ProxyGet](https://pkg.go.dev/k8s.io/client-go/kubernetes/typed/core/v1#ServiceExpansion) for the queries.

**Why do you want this feature?**

Users wouldn't need to setup port-forwarding, we can use the default kube-client and do it for them.

### PR DESCRIPTION
**Pull request by cPu1**
### Description

Closes #15 
Closes https://github.com/weaveworks/pctl/issues/21

### Checklist
- [ ] Added tests that cover your change
- [ ] Added/modified documentation as required (such as the `README.md`)
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] Added a `kind` label to the PR (e.g. `kind/feature`)

### CODE REVIEW

**Comment by cPu1** [comment]
The integration tests are failing because the code needs a Kubernetes control plane running in order to access the catalog service and proxy requests to it. If both the profiles and pctl code were in a single repo, we could've easily started a local environment (`make local-env`) before running the integration tests.

**Comment by aclevername** [comment]
Picking this up while Chetans away :smile:

File: Makefile [hunk 1]
```diff
@@ -3,10 +3,18 @@
 test: lint unit-test integration-test ## Run all tests
 
 unit-test: ## Run the unit tests
-	go test -count=1 ./pkg/...
+	ginkgo -r ./pkg
 
-integration-test: ## Run the integration tests
-	go test -count=1 ./tests/...
+integration-test: build local-env ## Run the integration tests
+	ginkgo -r ./tests/...
```
Comment (aclevername) [inline Makefile:9]: going to rename `integration` to acceptance and split out into separate github workflows in a follow up PR

File: cmd/pctl/main.go [hunk 2]
```diff
@@ -37,16 +41,16 @@ func searchCmd() *cli.Command {
 	return &cli.Command{
 		Name:      "search",
 		Usage:     "search for a profile",
-		UsageText: "pctl --catalog-url <URL> search <QUERY>",
+		UsageText: "pctl --kubeconfig=<kubeconfig-path> search <QUERY>",
 		Action: func(c *cli.Context) error {
-			searchName, catalogURL, err := parseArgs(c)
+			searchName, catalogClient, err := parseArgs(c)
 			if err != nil {
 				_ = cli.ShowCommandHelp(c, "show")
```
Comment (Callisto13) [inline cmd/pctl/main.go:48]:
```suggestion
				_ = cli.ShowCommandHelp(c, "search")
```
nothing to do with this PR, but can I sneak this in :p

**Comment by Callisto13** [comment]
@aclevername any docs or cmd help text needs updating for this?

File: README.md [hunk 3]
```diff
@@ -31,6 +31,7 @@ run the following to use pctl against it:
 
 1. Create your catalog, for example there is a `examples/profile-catalog-source.yaml` file in the profiles repo
 `kubectl apply -f profiles/examples/profile-catalog-source.yaml`
-1. In a separate terminal run `kubectl -n profiles-system port-forward <profiles-controller-pod-name> 8000:8000` to enable access to the API
-1. Run `pctl --catalog-url http://localhost:8000 search <query>` to search for your profile
-1. To see more detail of a profile, run `pctl --catalog-url http://localhost:8000 show <catalog-name>/<profile-name>`
+1. Ensure the current context in kubeconfig is set to the `profiles` cluster (`kubectl config current-context` should return `kind-profiles`)
```
Comment (Callisto13) [inline README.md:34]: is there a thing somewhere that returns a nice error if the user is pointed at a cluster which does not have the api running? (probs useful to have an error for if the api is not running in general, like they forgot to turn it on)

File: cmd/pctl/main.go (hunk 2 above)
Comment (aclevername) [reply cmd/pctl/main.go:48]: sure

File: README.md (hunk 3 above)
Comment (aclevername) [reply README.md:34]: that sounds reasonable, is it okay if thats in a follow up PR? I've only really touched 20% of this PR, so would like it to be merged before I fiddle around more :smile:

Comment (Callisto13) [reply README.md:34]: yep ofc, i will create a follow up thing 👍

**Comment by aclevername** [comment]
> @aclevername any docs or cmd help text needs updating for this?

Looks like the help text and readme have been updated 
```
./pctl
NAME:
   pctl - A cli tool for interacting with profiles

USAGE:
   pctl [global options] command [command options] [arguments...]

COMMANDS:
   search   search for a profile
   show     display information about a profile
   install  generate a profile subscription for a profile in a catalog
   help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --catalog-service-name value       Catalog Kubernetes Service name (default: "profiles-catalog-service")
   --catalog-service-port value       Catalog Kubernetes Service port (default: "8000")
   --catalog-service-namespace value  Catalog Kubernetes Service namespace (default: "profiles-system")
   --kubeconfig value                 Absolute path to the kubeconfig file (optional) (default: "/home/jake/.kube/config")
   --help, -h                         show help (default: false)
```

**Comment by Callisto13** [comment]
anything in the profiles repo?

**Comment by aclevername** [comment]
> anything in the profiles repo?

will do!

File: pkg/client/client.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
+package client
```
Comment (Callisto13) [inline pkg/client/client.go:1]: @aclevername are you doing tests for this file?

Comment (aclevername) [reply pkg/client/client.go:1]: missed this... will do

Comment (aclevername) [reply pkg/client/client.go:1]: added what I could

**Review by Callisto13 — APPROVED** [review]
lgtm 👍  lots of nice tidying up here, good job both 🚀

**Pull request merged**